dll/win32/shell32/CCopyToMenu.cpp

```diff
@@ -0,0 +1,116 @@
+#include "precomp.h"
+
+WINE_DEFAULT_DEBUG_CHANNEL(shell);
+
+CCopyToMenu::CCopyToMenu() :
+    m_pidlFolder(NULL),
+    m_idCmdFirst(0),
+    m_idCmdLast(0),
+    m_idCmdCopyTo(-1)
+{
+}
+
+CCopyToMenu::~CCopyToMenu()
+{
+    if (m_pidlFolder)
+        ILFree(m_pidlFolder);
+}
+
+HRESULT CCopyToMenu::DoCopyToFolder(LPCMINVOKECOMMANDINFO lpici)
+{
+    WCHAR wszPath[MAX_PATH];
+    HRESULT hr;
+
+    hr = SHGetPathFromIDListW(m_pidlFolder, wszPath);
+    if (FAILED_UNEXPECTEDLY(hr))
+        return hr;
+
+    BROWSEINFOW info = { lpici->hwnd };
+    info.pidlRoot = NULL;
+    info.lpszTitle = L"FIXME: This is a title";
+    info.ulFlags = BIF_RETURNONLYFSDIRS;
+    LPITEMIDLIST pidl = SHBrowseForFolder(&info);
```

Comment:
Why not use a `CComHeapPtr<ITEMIDLIST>` ?

Reply:
OK.